### PR TITLE
[prometheus-elasticsearch-exporter] Add ability to use values from secrets for basic-auth

### PR DESCRIPTION
**Pull request by abyth**
#### What this PR does / why we need it:

Alter the order of `Values.env` and `Values.extraEnvSecrets` to allow variable expansion to be able to use credentials from secrets for basic authentication. This is necessary as kubernetes only allows expansion if the variables are in the correct order so that environment variables from secrets have to be defined first.

> Note: Environment variables may reference each other, however ordering is important. Variables making use of others defined in the same context must come later in the list. Similarly, avoid circular references.

Source: https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/

E.g. the following `values.yaml` lead to the following deployment:

`values.yaml`
```yaml
...
extraEnvSecrets:
  ES_PASSWORD:
    secret: logs-es-elastic-user
    key: elastic
env:
  ES_URI: "https://elastic:$(ES_PASSWORD)@logs-es-http:9200/"
es:
  # Required to force usage of environment variable specified above
  uri: null
...
```

`deployment`
```yaml
...
      - command:
        - elasticsearch_exporter
        - --log.format=logfmt
        - --log.level=info
        - --es.all
        - --es.indices
        - --es.indices_settings
        - --es.shards
        - --es.snapshots
        - --es.timeout=30s
        - --es.ssl-skip-verify
        - --web.listen-address=:9108
        - --web.telemetry-path=/metrics
        env:
        - name: ES_PASSWORD
          valueFrom:
            secretKeyRef:
              key: elastic
              name: logs-es-elastic-user
        - name: ES_URI
          value: https://elastic:$(ES_PASSWORD)@logs-es-http:9200/
...
```

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-elasticsearch-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Elasticsearch stats exporter for Prometheus
 name: prometheus-elasticsearch-exporter
-version: 4.4.0
+version: 4.5.0
```
Comment (monotek) [inline charts/prometheus-elasticsearch-exporter/Chart.yaml:4]: update to 4.6.0 please

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-elasticsearch-exporter/Chart.yaml (hunk 1 above)
Comment (abyth) [reply charts/prometheus-elasticsearch-exporter/Chart.yaml]: bumped the version to `4.6.0` as requested

**Review by monotek — CHANGES_REQUESTED** [review]
Please sign commits to get the dco check working...

**Comment by abyth** [comment]
> Please sign commits to get the dco check working...



> Please sign commits to get the dco check working...

My bad.. i edited the commit message and forced a push

**Comment by abyth** [comment]
Is there anything i can do to progress the issue? Looks like the force-push did not really clear the change-request.

**Review by monotek — APPROVED** [review]

**Comment by monotek** [comment]
@desaintmartin please review :)

**Review by desaintmartin — APPROVED** [review]
Thanks!

**Pull request merged**